OpenSim/Simulation/Model/OffsetFrame.h

```diff
@@ -0,0 +1,159 @@
+#ifndef OPENSIM_OFFSET_FRAME_H_
+#define OPENSIM_OFFSET_FRAME_H_
+/* -------------------------------------------------------------------------- *
+ *                          OpenSim:  OffsetFrame.h                           
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    
+ * through the Warrior Web program.                                           
+ *                                                                            
+ * Copyright (c) 2005-2013 Stanford University and the Authors                
+ * Author(s): Matt DeMers Ajay Seth, Ayman Habib                              
+ *                                                                            
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         
+ *                                                                            
+ * Unless required by applicable law or agreed to in writing, software        
+ * distributed under the License is distributed on an "AS IS" BASIS,          
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        
+ * limitations under the License.                                             
+ * -------------------------------------------------------------------------- */
+
+// INCLUDE
+#include <OpenSim/Simulation/osimSimulationDLL.h>
+#include <OpenSim/Simulation/Model/Frame.h>
+#include <OpenSim/Simulation/Model/PhysicalFrame.h>
+
+namespace OpenSim {
+
+//=============================================================================
+//=============================================================================
+/**
+ * An OffsetFrame is a Frame whose transform (translation and orientation)
+ * with respect to another (parent) Frame is constant. It acts as an extension
+ * of the parent Frame type so that an OffsetFrame<Body>, for example,
+ * can be treated identically to a Body. This enables Frames to be filtered by
```

Comment:
> We don't want Frames that are not themselves bodies to be able to have inertia.

Ouch -- does seem like an OffsetFrame should be a kind of Frame, not a kind of Body (with mass props, geometry, etc.). I think we had it that way in one iteration of Frames. Something like:
```
                       Frame
                 /       |       \
     PhysicalFrame  ComputedFrame  OffsetFrame
     /        \
Body     OffsetPhysicalFrame      
```